packages/amplify_analytics_pinpoint/test/amplify_analytics_pinpoint_test.dart

```diff
@@ -26,95 +26,235 @@ void main() {
 
   TestWidgetsFlutterBinding.ensureInitialized();
 
-  setUp(() {
-    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
-      return true;
-    });
-  });
+  final eventName = 'test';
+  final userId = 'testUser';
+  final name = 'Darth Vader';
+  final email = 'vader@sith.com';
+  final plan = 'plan';
 
   tearDown(() {
     analyticsChannel.setMockMethodCallHandler(null);
   });
 
   // test sending basic events
-  test('recordEvent results in true', () async {
-    var event = AnalyticsEvent("test");
+  test('recordEvent correctly serializes input to method channel', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      expect(methodCall.arguments,
+          {'name': eventName, 'propertiesMap': {}, 'propertiesTypesMap': {}});
+    });
 
-    analytics.recordEvent(event: event);
+    var event = AnalyticsEvent(eventName);
+    expect(analytics.recordEvent(event: event), completes);
   });
 
-  test('recordEvent results in true', () async {
-    var event = AnalyticsEvent("test");
+  test('flushEvents executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.flushEvents(), completes);
+  });
 
-    event.properties.addBoolProperty("boolKey", true);
-    event.properties.addDoubleProperty("doubleKey", 10.0);
-    event.properties.addIntProperty("intKey", 10);
-    event.properties.addStringProperty("stringKey", "stringValue");
+  test('enable executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.enable(), completes);
+  });
 
-    analytics.recordEvent(event: event);
+  test('disable executes successfully', () async {
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    expect(analytics.disable(), completes);
   });
 
-  test('registerGlobalProperties results in true', () async {
-    var globalProperties = AnalyticsProperties();
+  test(
+      'unregisterGlobalProperties correctly serializes input to method channel',
+      () async {
+    final propertiesToUnregister = ['name'];
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      expect(methodCall.arguments, propertiesToUnregister);
+    });
+    expect(
+        analytics.unregisterGlobalProperties(
+            propertyNames: propertiesToUnregister),
+        completes);
+  });
+
+  test(
+      'recordEvent adds properties to event and correctly serializes input to method channel',
+      () async {
+    var event = AnalyticsEvent(eventName);
+    analyticsChannel.setMockMethodCallHandler((MethodCall methodCall) async {
+      expect(methodCall.arguments, {
+        'name': eventName,
+        'propertiesMap': {
```

Comment:
Not necessary, but if you have time perhaps we can move these maps to constants like you did for the event name and properties as well.

Reply:
Good call. I addressed in last commit and also tried to fully commit to the variables for others used in the file.